src/server/pps/cmds/cmds.go

```diff
@@ -235,6 +246,15 @@ $ {{alias}} -p foo -i bar@YYY`,
 				return errors.New("cannot set --output (-o) without --raw")
 			}
 
+			// To list jobs for all projects, user must be explicit about it.
+			// The --project filter takes precedence over everything else.
+			// By default use pfs.DefaultProjectName
+			projectsFilter := map[string]bool{project: true}
+			if allProjects {
+				projectsFilter = nil
+			} else if project == pfs.DefaultProjectName && project != pachCtx.GetProject() {
```

Comment:
Couldn’t this condition just be `project == pfs.DefaultProjectName`; if `project == pachCtx.GetProject()` then it doesn’t _hurt_ to reset `projectsFilter`, just wastes a few CPU cycles.